docs/DeveloperGuide.md

```diff
@@ -390,30 +390,70 @@ and simply import one after the other. As such if the user prefers updating thei
 the command handles that situation too by ignoring duplicate students and simply importing the new updates.
 
 
-### Mailing feature: mail-all
+### Mailing feature
 [<sub><sup>Back to top</sup></sub>](#table-of-contents)
 
+#### Aim of the feature
+This feature forms an integral part of automating communication for the user with the contacts present in the
+contact list. Mail commands allow the user to conveniently mail one or more students from the contact list.
+The main functionality of all the mail commands is that they would input the receiver's address for the user
+in the system default mail application, from where the user can continue on completing the mail on the external
+application.
+
 #### Implementation
 
 This Mailing feature enables the user to initiate the system
-default mail application (if present). In order to achieve this,
+default mail application (if present). In order to achieve this, 
 Java AWT (Abstract Window Toolkit) API is used.
 Below shows the important classes that were created:
 
-Logic:
-* MailIndexCommand (and its parser)
-* MailXCommand (and its parser)
-* MailAllCommand
+| Logic                  | Util     |
+|------------------------|----------|
+| MailIndexCommand       | MailUtil |
+| MailIndexCommandParser |          |
+| MailXCommand           |          |
+| MailXCommandParser     |          |
+| MailAllCommand         |          |
 
-Commons:
-* MailUtil
-
-MailUtil class contains the interaction of TAior with the desktop. All the commands
+MailUtil class contains the interaction of TAior with the desktop mail application. All the commands
```

Comment:
I think it might be better to give a concrete example of a command/method called?